gtsam/sfm/ShonanAveraging.h

```diff
@@ -151,6 +164,15 @@ class GTSAM_EXPORT ShonanAveraging {
     return measurements_[k];
   }
 
+  /// wrap factors with robust Huber loss
+  Measurements makeNoiseModelRobust(const Measurements& measurements) const {
+    Measurements robustMeasurements = measurements;
+    for (auto &measurement : robustMeasurements) {
+      measurement = BinaryMeasurement<Rot>(measurement, true);
```

Comment:
imperative, without any good reason. Please create a new Measurements variables, reserve space, and emplace_back :-)